chain/get_ancestors.go

```diff
@@ -51,38 +51,14 @@ func GetRecentAncestors(ctx context.Context, base types.TipSet, provider TipSetP
 		earliestAncestorHeight = types.NewBlockHeight(uint64(0))
 	}
 
-	// Step 1 -- gather all tipsets with a height greater than the earliest
-	// possible proving period start still in scope for the given head.
 	iterator := IterAncestors(ctx, provider, base)
-	provingPeriodAncestors, err := CollectTipSetsOfHeightAtLeast(ctx, iterator, earliestAncestorHeight)
-	if err != nil {
-		return nil, err
-	}
-	firstExtraRandomnessAncestorsCids, err := provingPeriodAncestors[len(provingPeriodAncestors)-1].Parents()
-	if err != nil {
-		return nil, err
-	}
-	// no parents means hit genesis so return the whole chain
-	if firstExtraRandomnessAncestorsCids.Len() == 0 {
-		return provingPeriodAncestors, nil
-	}
+	return CollectRecentTipSets(ctx, iterator, earliestAncestorHeight, lookback)
```

Comment:
Blocking: please don't merge together these two functions.  When I wrote "we no longer need two iterators" I didn't mean we don't want two functions operating over the iterator, I meant we don't need two calls to `IterAncestors`.  

We could have done this merging together of these functions before the change in #3026 (in fact I originally did this) but separated out the two functions to help readability and reusability.

Reply:
@ZenGround0 separated them. and rebase.